arangod/RocksDBEngine/RocksDBZkdIndex.cpp

```diff
@@ -424,11 +484,41 @@ void zkd::extractBoundsFromCondition(
     return false;
   };
 
+  auto const checkIsPrefixValue = [&](aql::AstNode* op, aql::AstNode* access,
+                                      aql::AstNode* other) -> bool {
+    TRI_ASSERT(op->type == aql::NODE_TYPE_OPERATOR_BINARY_EQ);
+
+    std::pair<aql::Variable const*, std::vector<basics::AttributeName>>
+        attributeData;
+    if (!access->isAttributeAccessForVariable(attributeData) ||
+        attributeData.first != reference) {
+      // this access is not referencing this collection
+      return false;
+    }
+
+    for (auto&& [idx, field] : enumerate(getSortedPrefixFields(index))) {
+      if (attributeData.second != field) {
+        continue;
+      }
+
+      auto [it, inserted] = extractedPrefix.emplace(idx, other);
+      TRI_ASSERT(inserted) << "duplicate access for " << attributeData.second;
```

Comment:
It is possible that the user specifies the same attribute multiple times in a filter condition by mistake, e.g.
`FILTER doc._from == 'abc' && doc._from == 'def'`, or `FILTER doc._from == 'abc' && doc._from == 'abc'`.
I think this is detected earlier on already, but maybe there are some cases under which such obviously senseless filter conditions will make it into this function. If that happens, I think we should handle it gracefully and not fail with an assertion failure (in maintainer mode).